python/tests/test_datasetslice.py

```diff
@@ -11,276 +11,363 @@
+                                         np.float64))
+        d1.coords['y'] = sc.Variable(['y'],
+                                     values=np.arange(M + 1).astype(
+                                         np.float64))
         arr1 = np.arange(N * M).reshape(N, M).astype(np.float64) + 1
-        d1[sc.Data.Value, "A"] = (['x', 'y'], arr1)
+        d1['A'] = sc.Variable(['x', 'y'], values=arr1)
         s1 = d1['x', 2:]
         s2 = copy.copy(s1)
         s3 = copy.deepcopy(s2)
         self.assertEqual(s1, s2)
         self.assertEqual(s3, s2)
         s2 *= s2
-        self.assertNotEqual(s1[sc.Data.Value, "A"], s2[sc.Data.Value, "A"])
-        self.assertNotEqual(s3[sc.Data.Value, "A"], s2[sc.Data.Value, "A"])
+        self.assertNotEqual(s1['A'], s2['A'])
+        self.assertNotEqual(s3['A'], s2['A'])
+
+    def test_copy_dataarrayview(self):
+        import copy
+        N = 6
+        M = 4
+        x = sc.Variable(['x'], values=np.arange(N + 1).astype(np.float64))
+        y = sc.Variable(['y'], values=np.arange(M + 1).astype(np.float64))
+        arr1 = np.arange(N * M).reshape(N, M).astype(np.float64) + 1
+        da = sc.DataArray(sc.Variable(['x', 'y'], values=arr1),
+                          coords={
+                              'x': x,
+                              'y': y
+                          })
+        s1 = da['x', 2:]
+        s2 = copy.copy(s1)
+        s3 = copy.deepcopy(s2)
+        self.assertEqual(s1, s2)
+        self.assertEqual(s3, s2)
+        s2 *= s2
+        self.assertNotEqual(s1, s2)
+        self.assertNotEqual(s3, s2)
 
     def _apply_test_op_rhs_ds_slice(self,
                                     op,
                                     a,
                                     b,
                                     data,
-                                    lh_var_name="a",
-                                    rh_var_name="b"):
+                                    lh_var_name='a',
+                                    rh_var_name='b'):
         # Assume numpy operations are correct as comparitor
         with np.errstate(invalid='ignore'):
-            op(data, b[sc.Data.Value, rh_var_name].numpy)
-        op(a, b)
-        # Desired nan comparisons
-        np.testing.assert_equal(a[sc.Data.Value, lh_var_name].numpy, data)
-
-    def _apply_test_op_rhs_variable(self,
-                                    op,
-                                    a,
-                                    b,
-                                    data,
-                                    lh_var_name="a",
-                                    rh_var_name="b"):
-        # Assume numpy operations are correct as comparitor
-        op(data, b.numpy)
+            op(data, b.data.values)
         op(a, b)
         # Desired nan comparisons
-        np.testing.assert_equal(a[sc.Data.Value, lh_var_name].numpy, data)
+        np.testing.assert_equal(a.data.values, data)
 
     def test_binary_slice_rhs_operations(self):
```

Comment:
At the developer meeting we can to the conclusion to focus Python tests on things that the Python bindings do. Everything else should be tested on the C++ side. This test and a number of others below appear to be retesting C++ code with very simple Python bindings. Shouldn't it be removed (or replaced by something that checks the effects of the bindings, e.g., returning a new `DataArray` in this case), after ensuring that thr equivalent is tested on the C++ side?

Reply:
Yes I think this is a necessary step. I focused too heavily on ensuring that all the old tests could be made to run (i.e. we didn't make any bad assumptions) since the tests were disabled. I'll recheck against all used bindings.